lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp

```diff
@@ -168,6 +168,267 @@ class ConvertAtenEmbeddingOp : public OpConversionPattern<AtenEmbeddingOp> {
+                                                        offsetsI, indicesIndex);
+                      Value offsetLessThanOrEqualToIndicesIndex =
+                          rewriter.create<arith::OrIOp>(
+                              loc, offsetLessThanIndicesIndex,
+                              offsetEqualToIndicesIndex);
+
+                      Value indicesIndexLessThanNextOffset =
+                          rewriter.create<arith::CmpIOp>(
+                              loc, arith::CmpIPredicate::slt, indicesIndex,
+                              nextOffset);
+
+                      Value indicesIndexWithinBounds =
+                          rewriter.create<arith::AndIOp>(
+                              loc, offsetLessThanOrEqualToIndicesIndex,
+                              indicesIndexLessThanNextOffset);
+
+                      SmallVector<Value> index_into_weight;
+                      index_into_weight.push_back(
+                          castIntToIndex(rewriter, loc, indexInIndices));
+                      index_into_weight.push_back(
+                          rewriter.create<linalg::IndexOp>(loc, /*value=*/2));
+                      Value weightElem = rewriter.create<tensor::ExtractOp>(
+                          loc, weight, index_into_weight);
+
+                      Value addResult = rewriter.create<arith::AddFOp>(
+                          loc, weightElem, initTensorElem);
+                      Value select = rewriter.create<arith::SelectOp>(
+                          loc, indicesIndexWithinBounds, addResult,
+                          initTensorElem);
+                      rewriter.create<linalg::YieldOp>(loc, select);
+                    })
+                .getResult(0);
+
+        // cast outputType.
+        auto resultType1 = typeConverter->convertType(op->getResult(0).getType());
+        Value initTensor_new =
+            rewriter.create<tensor::CastOp>(loc, resultType1, embeddingBagResult);
+
+        // offset2 tensor, this should be an empty tensor for the sum mode
+        SmallVector<Value> offset2size;
+        Type offsetElemTy = offsetsTy.getElementType();
+        Value zeroDim = rewriter.create<arith::ConstantIndexOp>(loc, /*value=*/0);
+        offset2size.push_back(zeroDim);
+        Value offset2 =
+            rewriter.create<linalg::InitTensorOp>(loc, offset2size, offsetElemTy);
+        auto resultType2 = typeConverter->convertType(op->getResult(1).getType());
+        Value offset2_new =
+            rewriter.create<tensor::CastOp>(loc, resultType2, offset2);
+
+        SmallVector<Value> offset_size = getTensorSizes(rewriter, loc, offsets);
+        // bagsize, vector of size offset with zeros, I think this is always just
+        // a vector of zeros in the sum mode
+        Value bagsize =
+            createZeroInitTensor(rewriter, loc, offset_size, offsetElemTy);
+        auto resultType3 = typeConverter->convertType(op->getResult(2).getType());
+        Value bagsize_new =
```

Comment:
Same comment as before: camelcase, and I would call these `bagSize` and `castedBagSize`, respectively